src/main/java/edu/ie3/util/TimeUtil.java

```diff
@@ -86,4 +93,47 @@ public String getDtfPattern() {
   public DateTimeFormatter getDateTimeFormatter() {
     return dateTimeFormatter;
   }
+
+  /**
+   * Determines the number of quarter hour of the day (starting with 0). The definition of quarter
+   * hour relates to something comparable than a right open interval. Example: 00:00:00 = 0 ...
+   * 00:14:59 = 0 00:15:00 = 1
+   *
+   * @param time Time, from which the number of quarter hour may be determined
+   * @return Number of the quarter hour of the day
+   */
+  public int getQuarterHourOfDay(ZonedDateTime time) {
+    return time.get(HOUR_OF_DAY) * 4 + time.get(MINUTE_OF_HOUR) / 15;
+  }
+
+  /**
+   * Calculate the difference between two given [[ZonedDateTime]]s and return the difference in
+   * seconds or {@link this#UNDEFINED_TIME} if the provided string is empty or equals to `undefined`
+   *
+   * @param startDateTime the start date time
+   * @param endDateTime the end date time
+   * @return the difference between the provided start and end date time in seconds
+   */
+  public double zonedDateTimeDifferenceInSeconds(String startDateTime, String endDateTime) {
```

Comment:
This is the only place, where `UNDEFINED_TIME` is used. Wouldn't it be better to throw an Exception and leave it open to the caller to handle this "fault"?

Additionally, wouldn't `long` be a better return type (as seconds is a countable amount)? Double would only bring the surplus, that you could treat milliseconds as well.

Reply:
you're right. And I would even go a step further and state, that it doesn't make sense at all to pass in a string in this method, bc the dev using this method should check themself about the validity of the `ZonedDateTime` instance that is passed in.  Hence I adapted the method accordingly.